src/Psl/Type/Internal/BoolType.php

```diff
@@ -34,7 +34,7 @@ public function coerce(mixed $value): bool
             return $value;
         }
 
-        if (0 === $value || '0' === $value) {
+        if (0 === $value || '0' === $value || '' === $value) {
```

Comment:
Isn't it a bit strange that only empty-string's will be allowed, but non-empty string result in a `CoercionException`?
Maybe an additional check for non-empty string could be added in the next if statement?

Reply:
Yea, this in my opinion is not desired, i think `CoercionException` for `''` is the correct behaviour.